components/prism-dataweave.js

```diff
@@ -0,0 +1,47 @@
+(function (Prism) {    
+    Prism.languages.dataweave = {	
+        'url': {
+            pattern:/\b(?:[A-Za-z])+:\/\/(?:[A-Za-z0-9/:\.\-_?=&])+|\burn:(?:[A-Za-z0-9:\-_\.?=&])+/
+        },
+        'property': {
+            pattern: /(?:[A-Za-z0-9_]+#)?(?:"(?:\\.|[^\\"\r\n])*"|[A-Za-z0-9_]+)(?=\s*[:@])/,
+            greedy: true
+        },
+        'string': {
+            pattern: /(["'`])(?:\\[\s\S]|(?!\1)[^\\])*\1/,
+            greedy: true
+        },
+        'mime-type': {
+            pattern: /(?:text|audio|video|application|multipart|image)\/(?:[A-Za-z-_+0-9])+/
+        },       
+        'date': {
+            pattern: /\|[^ ](?:[^\\|\r\n]|\\[^\r\n])+\|/,
```

Comment:
Why `[^ ]`? Is it really just the space character that is disallowed?
Also, is `||a|` valid then?

(Same for `regex`)

Reply:
I changed the regex to something clearer `/\|[\w\-:+-]+\|/` and added more tests where it shows all the possible values